File: Grid.cs
namespace Master_Thesis;

class Grid
{
    // Characteristic parameters of the grid and main (horizontal and vertical) axes of the building.
    private readonly int _numberVerticalAxes;
    private readonly int _numberHorizontalAxes;

    public Grid(int numVerAxes, int numHorAxes)
    {
        _numberVerticalAxes = numVerAxes;
        _numberHorizontalAxes = numHorAxes;
    }

    // Creating a list of axes intersection points related to the defined grid.
    public List<Point> SetGrid(int length, int width)
    {
         var spacingXdirection = length / (_numberVerticalAxes - 1);
         var spacingYdirection = width / (_numberHorizontalAxes - 1);

        List<Point> points = new List<Point>();

        for (int i = 0; i <= width; i += spacingYdirection)
        {
            for (int j = 0; j <= length; j += spacingXdirection)
            {
                var point = new Point(j, i);
                points.Add(point);
            }
        }
        return points;
    }

    public List<Point> GridOfAxes()
    {

        List<Point> points = new List<Point>();

        for (int i = 1; i <= _numberHorizontalAxes; i ++)
        {
            for (int j = 1; j <= _numberVerticalAxes; j ++)
            {
                var point = new Point(j, i);
                points.Add(point);
            }
        }
        return points;
    }
}

File: Python/Grid.py
from Point import Point

# Characteristic parameters of the grid and main (horizontal and vertical) axes of the building.
class Grid:
    def __init__(self, num_ver_axes, num_hor_axes):
        self.num_ver_axes = num_ver_axes
        self.num_hor_axes = num_hor_axes

    # Creating a list of axes intersection points related to the defined grid.
    def SetGrid(self, length, width):
        spacing_x_direction = length // (self.num_ver_axes -1)
        spacing_y_direction = width // (self.num_hor_axes -1)
        points = []
        i = 0
        j = 0
        while i <= width:
            while j <= length:
                point = Point(j,i)
                points.append(point)
                j += spacing_x_direction
            j = 0
            i += spacing_y_direction
        return points

